src/main/java/cz/it4i/fiji/datastore/register_service/DatasetRepository.java

```diff
@@ -44,7 +44,25 @@ public class DatasetRepository implements PanacheRepository<Dataset>,
 
 	@Inject
 	ApplicationConfiguration configuration;
+	public String findTypebyUUID(String uuid)
```

Comment:
Is it necessary as the type is part of a dataset?

Reply:
Probably yes, I don't know how else I would solve the choice of the right handler for writing/reading datasets.